### DISCUSSION TITLE
Analysis - Prometheus Metrics forever interval / until condition is met

### DISCUSSION THREAD

**Discussion by adlnc** [argoproj/argo-rollouts#3071] [Q&A]
Hi, 

I'm wondering if it's possible to configure Analysis to evaluate if a metric gets under a certain threshold and "ignore" the interval of measurement. Just looking at a query until the condition is true. 

For my use case, which might be illegitimate use of blue/green deployment, I'm trying to perform rollout with the use of AnalysisTemplate which is based on a **Prometheus metric**. 
As far as I understand, I can configure an `interval` of time during which the rollout is in progress. At the end of this interval, configured in the AnalysisTemplate, Rollouts checks if the `successCondition: result[0] <= 0.5` is met.
If the value returned from my prom query is, in my case below or eq to 0.5, it's successful and new version gets promoted otherwise, it's fail. 
If it is below at the beginning of the rollout but ends up above 0.5 after the interval duration, it fails.
Hopefully I'm understanding this right. From the tests that I've made, this is what I understood.

Now what if I want to ignore this interval of time, "running forever" and check every X min, until the successCondition is met. Would that be possible ?

I've tried to search for this kind of configuration but wasn't really successful. Maybe I'm twisting the purpose of this resource too much.
Also, I'm not sure I understand the `interval `and `count` parameters correctly. I find spec definition not so explicit about this. Maybe this is why I'm struggling here.

Any help would be much appreciated :)

Thanks

**Comment by juanfbl9307** [comment]
I have the same issue, the docs are not clear enough with interval and count or I’m missing something

**Reply by juanfbl9307** [reply]
I have found this:

// Interval refers to the Interval time window in Datadog (default: 5m). Not to be confused with the polling rate for…
--
optional string interval = 1;
 
// Interval defines an interval string (e.g. 30s, 5m, 1h) between each measurement.
 
optional string interval = 2;
 
// Count is the number of times to run the measurement. If both interval and count are omitted,
// the effective count is 1. If only interval is specified, metric runs indefinitely.
// If count > 1, interval must be specified.

https://github.com/argoproj/argo-rollouts/blob/00e39b114849010dd96221a8db441d58e860d4d0/pkg/apis/rollouts/v1alpha1/generated.proto#L581
